### PR TITLE
Implement viem Client

### PR DESCRIPTION
**Pull request by mmaurello**
### Description

Please explain the changes this PR addresses here.

### Checklist

- [ ] If this requires a documentation change, I have created a PR in [moonbeam-docs](https://github.com/PureStake/moonbeam-docs) repository.
- [x] If this requires it, I have updated the Readme
- [x] If necessary, I have updated the examples
- [x] I have verified if I need to create/update unit tests
- [x] I have verified if I need to create/update acceptance tests
- [x] If necessary, I have run acceptance tests on this branch in CI

### CODE REVIEW

File: packages/sdk/src/contract/contracts/Xtokens/Xtokens.ts [hunk 1]
```diff
@@ -1,26 +1,73 @@
...
 
-  readonly #signer: Signer;
-
-  constructor(config: ContractConfig, signer: Signer) {
+  constructor(config: ContractConfig, signer: Signer | WalletClient) {
     this.#config = config;
-    this.#signer = signer;
-    this.#contract = new Contract(this.address, abi, signer);
+
+    this.#client = isEthersSigner(signer)
```
Comment (ekenigs) [inline packages/sdk/src/contract/contracts/Xtokens/Xtokens.ts:28]: To simplify stuff I would try to use [getContract](https://viem.sh/docs/contract/getContract.html).

File: packages/sdk/src/sdk.interfaces.ts [hunk 2]
```diff
@@ -1,10 +1,14 @@
...
 import type { IKeyringPair } from '@polkadot/types/types';
-import type { Signer as EthersSigner } from 'ethers';
+import { Signer as EthersSigner } from 'ethers';
+import { WalletClient } from 'viem';
+
+export type EvmSigner = EthersSigner | WalletClient;
 
 export interface Signers {
-  ethersSigner: EthersSigner;
+  ethersSigner?: EthersSigner;
```
Comment (ekenigs) [inline packages/sdk/src/sdk.interfaces.ts:10]:
```suggestion
  /**
   * @deprecated ethersSigner - is deprecated and will be removed in v2, use evmSigner instead
   */
  ethersSigner?: EthersSigner;
```

**Review by ekenigs — CHANGES_REQUESTED** [review]
Please check my comments.

File: packages/sdk/package.json [hunk 3]
```diff
@@ -51,13 +51,14 @@
     "@moonbeam-network/xcm-config": "1.2.0",
     "@moonbeam-network/xcm-types": "1.0.1",
     "@moonbeam-network/xcm-utils": "1.0.2",
-    "big.js": "^6.2.1"
+    "big.js": "^6.2.1",
+    "ethers": "^5.7.2",
+    "viem": "^1.10.3"
```
[the remark this answers is not in the document]
Comment (mmaurello) [reply packages/sdk/package.json]: left them as `peerDependencies`

**Pull request merged**